var/spack/repos/builtin/packages/sirius/package.py

```diff
@@ -162,18 +168,30 @@ class Sirius(CMakePackage, CudaPackage, ROCmPackage):
         depends_on("dla-future +cuda", when="+cuda")
         depends_on("dla-future +rocm", when="+rocm")
 
+        conflicts("^pika@:0.22.1", when="+cuda")
+        conflicts("^pika@:0.22.1", when="+rocm")
+
     depends_on("rocblas", when="+rocm")
     depends_on("rocsolver", when="@7.5.0: +rocm")
 
+    # FindHIP cmake script only works for < 4.1
+    depends_on("hip@:4.0", when="@:7.2.0 +rocm")
```

Comment:
HIP versions < 5.1 were deprecated and have now been removed: https://github.com/spack/spack/pull/42676

Reply:
https://github.com/spack/spack/actions/runs/8854104193/job/24316430409?pr=43872#step:6:53

Reply:
Thanks, I've made the `depends_on("hip@:4")` a conflict and tried to fix the other complaints of the linter.